src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -132,6 +142,242 @@ public async Task SetUseSuggestionModeAsync(bool value, CancellationToken cancel
+            var viewAdapter = editorAdaptersFactoryService.GetViewAdapter(textView);
+            Assumes.Present(viewAdapter);
+
+            // Make sure we have the top pane
+            //
+            // The docs are wrong. When a secondary view exists, it is the secondary view which is on top. The primary
+            // view is only on top when there is no secondary view.
+            var codeWindow = TryGetCodeWindow(viewAdapter);
+            Assumes.Present(codeWindow);
+
+            if (ErrorHandler.Succeeded(codeWindow.GetSecondaryView(out var secondaryViewAdapter)))
+            {
+                viewAdapter = secondaryViewAdapter;
+            }
+
+            var textViewHost = editorAdaptersFactoryService.GetWpfTextViewHost(viewAdapter);
+            Assumes.Present(textViewHost);
+
+            var dropDownMargin = textViewHost.GetTextViewMargin("DropDownMargin");
+            if (dropDownMargin != null)
+            {
+                return ((Decorator)dropDownMargin.VisualElement).Child;
+            }
+
+            return null;
+
+            static IVsCodeWindow? TryGetCodeWindow(IVsTextView textView)
+            {
+                if (textView is not IObjectWithSite objectWithSite)
+                {
+                    return null;
+                }
+
+                var riid = typeof(IOleServiceProvider).GUID;
+                objectWithSite.GetSite(ref riid, out var ppvSite);
+                if (ppvSite == IntPtr.Zero)
+                {
+                    return null;
+                }
+
+                IOleServiceProvider? oleServiceProvider = null;
+                try
+                {
+                    oleServiceProvider = Marshal.GetObjectForIUnknown(ppvSite) as IOleServiceProvider;
+                }
+                finally
+                {
+                    Marshal.Release(ppvSite);
+                }
+
+                if (oleServiceProvider == null)
+                {
+                    return null;
+                }
+
+                var guidService = typeof(SVsWindowFrame).GUID;
+                riid = typeof(IVsWindowFrame).GUID;
+                if (ErrorHandler.Failed(oleServiceProvider.QueryService(ref guidService, ref riid, out var ppvObject)) || ppvObject == IntPtr.Zero)
+                {
+                    return null;
+                }
+
+                IVsWindowFrame? frame = null;
+                try
+                {
+                    frame = (IVsWindowFrame)Marshal.GetObjectForIUnknown(ppvObject);
+                }
+                finally
+                {
+                    Marshal.Release(ppvObject);
+                }
+
+                riid = typeof(IVsCodeWindow).GUID;
+                if (ErrorHandler.Failed(frame.QueryViewInterface(ref riid, out ppvObject)) || ppvObject == IntPtr.Zero)
+                {
+                    return null;
```

Comment:
would it be useful at all to have logs on failure cases here, or do the dumps typically contain all the information we'd need to verify which of these error cases we're hitting?

Reply:
➡️ It definitely wouldn't hurt, especially considering the next like after calling `TryGetCodeWindow` is always `Assumes.Present`. For now I'm OK keeping this form for two reasons:

1. This code historically has not failed (copied from older harness)
2. In the event of failure, we'll get a message saying IVsCodeWindow was expected but not available, and at that point it would be easy to update the implementation to better explain the point of failure

Reply:
> In the event of failure, we'll get a message saying IVsCodeWindow was expected but not available, and at that point it would be easy to update the implementation to better explain the point of failure

taht wfm.  it's generally waht i've done when there are issues and the logs aren't sufficient.